### DISCUSSION TITLE
Add support for Finance::Quote

### DISCUSSION THREAD

**Discussion by ollieluvic** [ghostfolio/ghostfolio#2465] [Ideas]
First and foremost, I commend the Ghostfolio devs on the amazing work you guys have been doing. I’d also like to apologize if there’s something wrong in my writings as I’m neither an expert in finance/investing nor am very acquainted with the Ghostfolio codebase, having only scanned through some of the source code of the data providers.

## Motivation

Some region-specific assets are not accessible from Yahoo Finance, an example being Brazillian mutual funds like [PACTIPC:BZ](https://www.bloomberg.com/quote/PACTIPC:BZ), which quotes seem to only be accessible through Bloomberg. In addition to Brazillian mutual funds and treasury bonds, it seems that people in other regions of the world could benefit from this change as that also seems to be the case for Swidish assets as evidenced by issue [#1901](https://github.com/ghostfolio/ghostfolio/issues/1901). I believe this could also allow support for US Treasury Bonds, which seems to not exist today as is suggested by a comment on issue [#1352]( https://github.com/ghostfolio/ghostfolio/issues/1352#issuecomment-1725752177).

My rationale is that by integrating Finance::Quote, the application would hugely increase the list of supported assets as well as reducing the work needed to implement and maintain new data providers allowing the GF devs to focus on features/issues that could have a bigger value yield to the application.

## What is Finance::Quote?

I assume the devs might already be familiar with it, but for the sake of making this discussion more complete, I'm including a quick introduction into the module. 

Finance::Quote is a Perl module for getting stock quotes from various internet sources all over the world. It is the module utilized by GnuCash for fetching currency and asset quotes from the Internet. More at their [README page](https://github.com/finance-quote/finance-quote/tree/master).

## What would it take?

At a first glance, there would be some development effort that’d need to take place to integrate the Perl module into the Typescript codebase, possibly an adapter module. Other than that it would probably warrant some changes to the UI (and obviously the back-end too) so that users have the option of selecting their preffered quote source for an asset, somewhat like in the GnuCash screenshot below.

![image](https://github.com/ghostfolio/ghostfolio/assets/12973202/8bb93a19-fd1d-4873-be7e-ad0ac7a4762d)

Another option that comes to mind would be to dynamically handle the source selection when fetching the quotes, I believe, however, that it could be prone to break due to some divergence in the naming of an asset across sources, though I’m not sure if that would be a thing.

---
I’d be willing to contribute some work into this integration if the maintainers feel this integration would be a good direction for the project. I mostly have experience in Front-End Development and ReactJS, with some Back-End experience (mainly building APIs with NestJS).

**Comment by dtslvr** [comment]
Hi @luvicsoura

Thank you for your explanations. We are quite reserved when it comes to integrating new data providers. However, with the help of the data source `MANUAL` it is basically possible to retrieve data from any URL.

From an architecture perspective, my suggestion would be that you build a small NestJS API that wraps `Finance::Quote` and can then be called periodically by Ghostfolio. What do you think about this?
